tests/Tests.AzureAppConfiguration/RefreshTests.cs

```diff
@@ -101,10 +121,41 @@ public void RefreshTests_RefreshIsSkippedIfCacheIsNotExpired()
                 .AddAzureAppConfiguration(options =>
                 {
                     options.Client = mockClient.Object;
-                    options.Select("TestKey*");
                     options.ConfigureRefresh(refreshOptions =>
                     {
-                        refreshOptions.Register("TestKey1")
+                        refreshOptions.Register("TestKey1", "label")
+                            .SetCacheExpiration(TimeSpan.FromSeconds(10));
+                    });
+
+                    refresher = options.GetRefresher();
+                })
+                .Build();
+
+            Assert.Equal("TestValue1", config["TestKey1"]);
+            FirstKeyValue.Value = "newValue1";
+
+            // Wait for some time but not enough to let the cache expire
+            Thread.Sleep(5000);
+
+            refresher.RefreshAsync().Wait();
+
+            Assert.Equal("TestValue1", config["TestKey1"]);
+        }
+
+        [Fact]
+        public void RefreshTests_RefreshIsSkippedIfKvNotInSelectAndCacheIsNotExpired()
```

Comment:
The test updates in this PR got really big. Can we shorten it to just this method and GetMockConfigurationClientSelectKeyLabel?

Reply:
I found that some refresh scenario tests were actually not testing the intended functionality because of the following reasons:

- We were selecting/registering KVs based on just key name, whereas all the KVs in mock data also had a label. That led to unintentionally registering/selecting non-existent keys with null labels, and hence the test took a different code path than expected. 
For Eg: 

    - When `RefreshAsync `call triggers `RefreshIndividualKeyValues`, instead of calling `GetKeyValueChange `and processing the "changes", we load the setting again with `GetConfigurationSettingAsync`. That misses the intention of testing the `GetKeyValueChange `method.

    - When `LoadKeyValuesRegisteredForRefresh `is called, we should skip loading KV that have already loaded as part of `Select` clause. But since we registered a KV without its label, it's treated as a new KV and loaded again. This is actually why `RefreshTests_RefreshIsSkippedIfCacheIsNotExpired` was passing instead of catching the bug addressed in this PR.

- We were always returning the entire mock data collection of KV, regardless of what is present in the `Select` clause. This led to unintentional behavior in tests like `RefreshTests_RefreshRegisteredKeysAreLoadedOnStartup_DefaultUseQuery` and `RefreshTests_RefreshRegisteredKeysAreLoadedOnStartup_CustomUseQuery`. These tests would always pass regardless of `ConfigureRefresh `because the `Select` clause (or a lack thereof), ensures that all KVs from mock data collection are loaded.


To fix this, I have made the following changes to these tests:

1. I've explicitly added labels to every key that is registered for refresh. 
2. Because of # 1, I also had to register a new mock client setup for `GetConfigurationSettingAsync(ConfigurationSetting setting, bool cond, CancellationToken ct)` method.
3. Because of # 2, I was able to refactor all the tests that had the same mock client setup and replace the duplicated code with the shared `GetMockConfigurationClient()` method.
4. I've also added a new method `GetMockConfigurationClientSelectKeyLabel` for the tests that absolutely rely on loading only the specified KV with `Select` clause.

The only reason I included all these changes in this PR is because I wanted to fix the test that is related to this PR (`RefreshTests_RefreshIsSkippedIfCacheIsNotExpired`) by adding the correct label for refresh registered KV, and that small change triggered all the other changes that are mentioned above. 

But I understand that these are a lot of changes for one PR which are unrelated to the bug fix, so in this PR, I can add the new test `RefreshTests_RefreshIsSkippedIfKvNotInSelectAndCacheIsNotExpired` and `GetMockConfigurationClientSelectKeyLabel`, and create a new PR for all the other test changes.

Reply:
That sounds like a good idea. Thanks Avani.